user/signer.go

```diff
@@ -0,0 +1,70 @@
+package user
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	neofsecdsa "github.com/nspcc-dev/neofs-sdk-go/crypto/ecdsa"
+)
+
+// Signer is an interface of entities that can be used for signing operations
+// in NeoFS. It is the same as [neofscrypto.Signer], but has on extra method to retrieve [ID].
+type Signer interface {
+	neofscrypto.Signer
+
+	UserID() ID
+}
+
+// SignerRFC6979 wraps [ecdsa.PrivateKey] and represents signer based on deterministic
+// ECDSA with SHA-256 hashing (RFC 6979). Provides [Signer] and [neofscrypto.Signer] interfaces.
```

Comment:
I think just `Signer` is enough to mention here, everything else is its details.

Reply:
Updated